src/interface/swarm.cpp

```diff
@@ -282,58 +288,52 @@ void Swarm::setPoolMax(const std::int64_t nmax_pool) {
   nmax_pool_ = nmax_pool;
 }
 
-ParArray1D<bool> Swarm::AddEmptyParticles(const int num_to_add,
-                                          ParArrayND<int> &new_indices) {
-  if (num_to_add <= 0) {
-    new_indices = ParArrayND<int>();
-    return ParArray1D<bool>();
-  }
-
-  while (free_indices_.size() < num_to_add) {
-    increasePoolMax();
-  }
+NewParticlesContext Swarm::AddEmptyParticles(const int num_to_add) {
+  PARTHENON_DEBUG_REQUIRE(num_to_add >= 0, "Cannot add negative numbers of particles!");
```

Comment:
Does it even makes sense to allow for `num_to_add == 0`?

Reply:
I think this case can easily appear in downstream codes, if say you have some function that samples a certain number of particles on each meshblock according to some function, that function often says to sample zero particles in certain regions. If we prevent that case here, it requires extra logic on the downstream code side.